CMakeLists.txt

```diff
@@ -256,6 +256,8 @@ if(UNIX)
         add_compile_definitions("_POSIX_C_SOURCE=200112L")
     endif()
     cmake_pop_check_state()
+    cmake_reset_check_state()
+    check_symbol_exists(explicit_bzero "string.h" REALM_HAVE_EXPLICIT_BZERO)
```

Comment:
This needs to be just before the `cmake_pop_check_state()` call - CMake checks should ideally always be between state check push and pop, even if this particular check isn't modifying any of the `CMAKE_REQUIRED_*` variables. Here we just call `reset_check_state()` in between checks so we don't push and pop for every check.